make_prg/subcommands/update.py

```diff
@@ -0,0 +1,187 @@
+    )
+    subparser_update_prg.set_defaults(func=run)
+
+    return subparser_update_prg
+
+
+@dataclass
+class UpdateSharedData:
+    denovo_variants_db: DenovoVariantsDB
+    aligner: MSAAligner
+
+
+def update(locus_name: str):
+    global options
+    prg_builder_zip_db = PrgBuilderZipDatabase(options.update_DS)
+    prg_builder_zip_db.load()
+    prg_builder_for_locus = prg_builder_zip_db.get_PrgBuilder(locus_name)
+
+    global update_shared_data
+    prg_builder_for_locus.aligner = update_shared_data.aligner
+    update_data_list = update_shared_data.denovo_variants_db.locus_name_to_update_data.get(locus_name, [])
+    nb_of_variants_sucessfully_updated = 0
+    nb_of_variants_with_failed_update = 0
+
+    we_have_variants = len(update_data_list) > 0
+    if we_have_variants:
+        logger.debug(f"Updating {locus_name} ...")
+
+        leaves_to_update = set()
+        for update_data in update_data_list:
+            try:
+                prg_builder_tree_node = prg_builder_for_locus.get_node_given_interval(
+                    update_data.ml_path_node_key
+                )
+                prg_builder_tree_node.add_data_to_batch_update(
+                    update_data
+                )
+                leaves_to_update.add(prg_builder_tree_node)
+                nb_of_variants_sucessfully_updated += 1
+            except LeafNotFoundException as exc:
+                logger.warning(f"Failed finding leaf: {exc}")
+                nb_of_variants_with_failed_update += 1
+
+        # update the modified leaves
+        # Note: the sorted() is needed for determinism so that we can compare indexes and test outputs
+        for leaf in sorted(leaves_to_update, key=lambda node: node.id):
+            leaf.batch_update()
+        logger.debug(
+            f"Updated {locus_name}: {nb_of_variants_sucessfully_updated} denovo sequences added!"
+        )
+    else:
+        logger.debug(f"{locus_name} has no new variants, no update needed")
+
+    # regenerate PRG
+    temp_dir = io_utils.get_temp_dir_for_multiprocess(options.temp_dir)
+    locus_prefix = temp_dir / locus_name
+
+    logger.info(f"Writing output files of locus {locus_name}")
+    prg = prg_builder_for_locus.build_prg()
+    prg_builder_for_locus.write_prg(str(locus_prefix), prg)
+    gfa.GFA_Output.write_gfa(str(locus_prefix), prg)
+    prg_builder_for_locus.serialize(f"{locus_prefix}.pickle")
+    with open(f"{locus_prefix}.stats", "w") as stats_filehandler:
+        print(
+            f"{locus_name} {nb_of_variants_sucessfully_updated} {nb_of_variants_with_failed_update}",
+            file=stats_filehandler,
+        )
+    if options.output_graphs:
+        prg_builder_for_locus.output_debug_graphs(Path(options.output_prefix + "_debug_graphs"))
+
+
+def run(cl_options):
+    global options
+    options = cl_options
+    global update_shared_data
+
+    if io_utils.output_files_already_exist(options.output_prefix):
```

Comment:
Again, a force option would be ideal

Reply:
Will be done in next commit!

Reply:
Done in https://github.com/iqbal-lab-org/make_prg/pull/35/commits/0cda733793f603cd76ad146e62e839bb53844e3d , please check. Also added 5 integration tests to test well this parameter, but these tests will be described only in a future PR